grails-app/services/org/broadinstitute/orsp/StatusEventService.groovy

```diff
@@ -56,69 +61,50 @@ class StatusEventService {
         // Terminal Event Case:
         // If the event is a terminal event (i.e. completed, abandoned, etc.) then the duration is
         // calculated from the first event to the terminal state event.
-        List<StatusEventDTO> eventDTOs = statusEvents.
-                withIndex().
+        List<StatusEventDTO> eventDTOs = statusEvents?.
+                withIndex()?.
                 collect { Event event, int i ->
-                    Period duration
+                    Long durationTime
```

Comment:
I'm curious about why this is better than `Period`. It could be more efficient, I just don't know. I do remember that `Period` handled some date conversions for me that were problematic at the time, but I don't remember what those were.

Reply:
As it was before, it used Period as way to build and return the string "xx Years and nn months and yy days" directly from groovy method.
Since we need to sort this in bootstrap table, it was easier to return milliseconds and parse each value as string using javascript Date.
If you consider it would be better to use Java 8 object i could use `ChronoUnit.between` method. [Doc](https://docs.oracle.com/javase/tutorial/datetime/iso/period.html).

Reply:
Thanks, makes sense. I'm fine with the current approach.